xbmc/addons/kodi-addon-dev-kit/include/kodi/addon-instance/PVRClient.h

```diff
@@ -0,0 +1,4763 @@
+typedef struct PVR_ADDON_CAPABILITIES
+{
+  bool bSupportsEPG;
+  bool bSupportsEPGEdl;
+  bool bSupportsTV;
+  bool bSupportsRadio;
+  bool bSupportsRecordings;
+  bool bSupportsRecordingsUndelete;
+  bool bSupportsTimers;
+  bool bSupportsChannelGroups;
+  bool bSupportsChannelScan;
+  bool bSupportsChannelSettings;
+  bool bHandlesInputStream;
+  bool bHandlesDemuxing;
+  bool bSupportsRecordingPlayCount;
+  bool bSupportsLastPlayedPosition;
+  bool bSupportsRecordingEdl;
+  bool bSupportsRecordingsRename;
+  bool bSupportsRecordingsLifetimeChange;
+  bool bSupportsDescrambleInfo;
+  bool bSupportsAsyncEPGTransfer;
+
+  /*!
+    * @brief **required**\n Count of possible values for PVR_RECORDING.iLifetime.
+    * 0 means lifetime is not supported for recordings or no own value
+    * definition wanted, but to use Kodi defaults of 1..365.
+    */
+  unsigned int iRecordingsLifetimesSize;
+
+  /*!
+    * @brief **optional**\n Array containing the possible values for
+    * PVR_RECORDING.iLifetime. Must be filled if iLifetimesSize > 0
+    */
+  PVR_ATTRIBUTE_INT_VALUE recordingsLifetimeValues[PVR_ADDON_ATTRIBUTE_VALUES_ARRAY_SIZE];
+} ATTRIBUTE_PACKED PVR_ADDON_CAPABILITIES;
+
+//==============================================================================
+/// \ingroup cpp_kodi_addon_pvr_Defs
+/// @brief **PVR add-on capabilities**
+///
+/// If a capability is set to **true**, then the corresponding methods from
+/// @ref kodi::addon::CInstancePVRClient need to be implemented.
+///
+/// Used on @ref kodi::addon::CInstancePVRClient::GetAddonCapabilities().
+///
+//@{
+struct AddonPVRCapabilities
+{
+  /*! \cond PRIVATE */
+  explicit AddonPVRCapabilities() = default;
+  AddonPVRCapabilities(const PVR_ADDON_CAPABILITIES* capabilities)
+  {
+    supportsEPG = capabilities->bSupportsEPG;
+    supportsEPGEdl = capabilities->bSupportsEPGEdl;
+    supportsTV = capabilities->bSupportsTV;
+    supportsRadio = capabilities->bSupportsRadio;
+    supportsRecordings = capabilities->bSupportsRecordings;
+    supportsRecordingsUndelete = capabilities->bSupportsRecordingsUndelete;
+    supportsTimers = capabilities->bSupportsTimers;
+    supportsChannelGroups = capabilities->bSupportsChannelGroups;
+    supportsChannelScan = capabilities->bSupportsChannelScan;
+    supportsChannelSettings = capabilities->bSupportsChannelSettings;
+    handlesInputStream = capabilities->bHandlesInputStream;
+    handlesDemuxing = capabilities->bHandlesDemuxing;
+    supportsRecordingPlayCount = capabilities->bSupportsRecordingPlayCount;
+    supportsLastPlayedPosition = capabilities->bSupportsLastPlayedPosition;
+    supportsRecordingEdl = capabilities->bSupportsRecordingEdl;
+    supportsRecordingsRename = capabilities->bSupportsRecordingsRename;
+    supportsRecordingsLifetimeChange = capabilities->bSupportsRecordingsLifetimeChange;
+    supportsDescrambleInfo = capabilities->bSupportsDescrambleInfo;
+    supportsAsyncEPGTransfer = capabilities->bSupportsAsyncEPGTransfer;
```

Comment:
You should prefer member init list over init in ctor body. Here and everywhere else in this file.

Reply:
How exact you mean?

Reply:
https://en.cppreference.com/w/cpp/language/initializer_list